### PR TITLE
Add various REST API docs (emails, files, marketing messages, notes, pages, roles)

### PR DESCRIPTION
**Pull request by dennisameling**
Had a productive flight tonight! Another large chunk of https://github.com/mautic/developer-documentation-new/issues/97 done!

<a href="https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/139"></a>

### CODE REVIEW

**Review by escopecz — DISMISSED** [review]
Looks good to me 👍 Thanks Dennis! You should fly more often 😄

File: docs/rest_api/notes.rst [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+   );
+
+   // Create new a note of ID 1 isn't found?
+   $createIfNotFound = true;
+
+   $note = $noteApi->edit($id, $data, $createIfNotFound);
+
+Edit a new note. Note that this supports PUT or PATCH depending on the desired behavior.
+
+**PUT** creates a note if the given ID doesn't exist and clears all the note information, adds the information from the request.
```
Comment (RCheesley) [inline docs/rest_api/notes.rst:268]:
```suggestion
**PUT** creates a Note if the given ID doesn't exist and clears all the Note information, adds the information from the request.
```

File: docs/rest_api/notes.rst [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+
+**PATCH** fails if the note with the given ID doesn't exist and updates the note field values with the values from the request.
```
Comment (RCheesley) [inline docs/rest_api/notes.rst:270]:
```suggestion
**PATCH** fails if the Note with the given ID doesn't exist and updates the Note field values with the values from the request.
```

File: docs/rest_api/notes.rst [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+
+.. vale off
+
+**HTTP Request**
+
+.. vale on
+
+To edit a note and return a 404 if the note isn't found:
```
Comment (RCheesley) [inline docs/rest_api/notes.rst:278]:
```suggestion
To edit a Note and return a 404 if the Note isn't found:
```

File: docs/rest_api/notes.rst [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+
+``PATCH /notes/ID/edit``
+
+To edit a note and create a new one if the note isn't found:
```
Comment (RCheesley) [inline docs/rest_api/notes.rst:282]:
```suggestion
To edit a Note and create a new one if the Note isn't found:
```

File: docs/rest_api/notes.rst [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+     - Description
+   * - ``text``
+     - string
+     - Note text
+   * - ``type``
+     - string
+     - Note type
+   * - ``datetime``
+     - ``datetime``
+     - Date and time related to the note.
```
Comment (RCheesley) [inline docs/rest_api/notes.rst:302]:
```suggestion
     - Date and time related to the Note.
```

File: docs/rest_api/notes.rst [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+
+
+**Response**
+
+If ``PUT``, the expected response code is ``200`` if editing the note or ``201`` if created.
```
Comment (RCheesley) [inline docs/rest_api/notes.rst:307]:
```suggestion
If ``PUT``, the expected response code is ``200`` if editing the Note or ``201`` if created.
```

File: docs/rest_api/notes.rst [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+
+.. vale on
+
+.. code-block:: php
+
+   <?php
+
+   $note = $noteApi->delete($id);
+
+Delete a note.
```
Comment (RCheesley) [inline docs/rest_api/notes.rst:328]:
```suggestion
Delete a Note.
```

File: docs/rest_api/pages.rst [hunk 8]
```diff
@@ -0,0 +1,462 @@
...
+     - Total Landing Page hit count
+   * - ``uniqueHits``
+     - int
+     - Unique Landing Page hit count
+   * - ``revision``
+     - int
+     - Landing Page revision
+   * - ``metaDescription``
+     - string
+     - Meta description for the Landing Page's ``<head>``
```
Comment (RCheesley) [inline docs/rest_api/pages.rst:155]: Does this need to be plural? If so should the others?

File: docs/rest_api/roles.rst [hunk 9]
```diff
@@ -0,0 +1,387 @@
...
+.. code-block:: php
+
+   <?php
+   use Mautic\MauticApi;
+   use Mautic\Auth\ApiAuth;
+
+   // ...
+   $initAuth = new ApiAuth();
+   $auth     = $initAuth->newAuth($settings);
+   $apiUrl   = "https://your-mautic.com";
```
Comment (RCheesley) [inline docs/rest_api/roles.rst:19]:
```suggestion
   $apiUrl   = "https://mautic.example.com";
```

File: docs/rest_api/roles.rst [hunk 10]
```diff
@@ -0,0 +1,387 @@
...
+
+.. vale on
+
+.. code-block:: php
+
+   <?php 
+
+   $data = array(
+       'name' => 'API test role',
+       'description' => 'created via AIP',
```
Comment (RCheesley) [inline docs/rest_api/roles.rst:227]:
```suggestion
       'description' => 'created via API',
```

File: docs/rest_api/roles.rst [hunk 11]
```diff
@@ -0,0 +1,387 @@
...
+.. vale on
+
+.. code-block:: php
+
+   <?php
+
+   $id   = 1;
+   $data = array(
+       'name' => 'API test role',
+       'description' => 'created via AIP',
```
Comment (RCheesley) [inline docs/rest_api/roles.rst:292]:
```suggestion
       'description' => 'created via API',
```

File: docs/rest_api/roles.rst [hunk 12]
```diff
@@ -0,0 +1,387 @@
...
+           ),
+       )
+   );
+
+   // Create new a Role of ID 1 isn't found?
+   $createIfNotFound = true;
+
+   $role = $roleApi->edit($id, $data, $createIfNotFound);
+
+Edit a new Role. Role that this supports PUT or PATCH depending on the desired behavior.
```
Comment (RCheesley) [inline docs/rest_api/roles.rst:307]:
```suggestion
Edit a new Role. Note that this supports PUT or PATCH depending on the desired behavior.
```

File: docs/rest_api/users.rst [hunk 13]
```diff
@@ -0,0 +1,541 @@
...
+   $data = array(
+       'lastName' => 'Doeboe',
+   );
+
+   // Create new a User of ID 1 isn't found?
+   $createIfNotFound = true;
+
+   $user = $userApi->edit($id, $data, $createIfNotFound);
+
+Edit a new User. User that this supports PUT or PATCH depending on the desired behavior.
```
Comment (RCheesley) [inline docs/rest_api/users.rst:380]:
```suggestion
Edit a new User. Note that this supports PUT or PATCH depending on the desired behavior.
```

File: docs/rest_api/users.rst [hunk 14]
```diff
@@ -0,0 +1,541 @@
...
+.. code-block:: json
+
+   {
+     "user:users:create":true,
+     "user:users:edit":true
+   }
+
+**Properties**
+
+array of requested permissions of string in case of just one
```
Comment (RCheesley) [inline docs/rest_api/users.rst:541]:
```suggestion
An array of requested permissions, or a string where returning only one value.
```

I think this is what you mean but please check to be sure of the intended context!

**Review by RCheesley — CHANGES_REQUESTED** [review]
Thanks so much for this @dennisameling ! A few typos / suggestions, and a couple where I am not 100% sure on the intention so please double check!

File: docs/rest_api/files.rst [hunk 15]
```diff
@@ -0,0 +1,187 @@
+Files
+#####
+
+This endpoint is useful for working with files of images and Assets.
```
[the remark this answers is not in the document]
Comment (dennisameling) [reply docs/rest_api/files.rst:4]: Good question! This is literally copy-pasted from the old docs. I was just laser-focused on getting everything converted from markdown to RST and fixing all the Vale errors/warnings 😅 Looking at the docs, it seems that Files can indeed both be images or "assets". At this point I'm not sure what exactly is meant by those, but I think this endpoint has very similar behavior as Mautic's media uploader. I'll do some testing tomorrow.

[the remark this answers is not in the document]
Comment (escopecz) [reply docs/rest_api/files.rst:4]: Yea, this endpoint allows to upload files. There are 2 places in Mautic where users can upload files:
1. Images for the email/page builder
2. Assets that users can download.

This endpoint allows to upload files to both places.

[the remark this answers is not in the document]
Comment (RCheesley) [reply docs/rest_api/files.rst:4]: Yeah so I would change it to be:

```suggestion
This endpoint is useful for working with files, which might include images and Assets.
```

File: docs/rest_api/pages.rst (hunk 8 above)
Comment (dennisameling) [reply docs/rest_api/pages.rst:155]: Nope, singular in this case. This API endpoint allows for retrieving a single Landing Page 😊 To improve clarity a bit, I'll change this to:

> Meta description which is put in the Landing Page's ``<head>`` element.

File: docs/rest_api/users.rst (hunk 14 above)
Comment (dennisameling) [reply docs/rest_api/users.rst:541]: I'll test this endpoint in a sec and will update the description here. Thanks for noticing!

**Comment by dennisameling** [comment]
The only remaining error [is a known issue](https://github.com/mautic/developer-documentation-new/pull/139#discussion_r1274127410). Going to merge this now 👍🏼

**Pull request merged**